website/usersearch.php

```diff
@@ -58,7 +58,15 @@
 	if ($users->csrfCheck()) {
 		$csrfOk = true;
 	}
-	$userlist = $psdb->query("SELECT `username`, `userid`, `banstate` FROM `ntbb_users` WHERE `ip` = ?", [$ip]);
+	$isRange = (substr($ip, -1) === "*");
+	if ($isRange) {
+		$ip = substr($ip, 0, strlen($ip) - 1);
+	}
+	$where = $isRange ? "`ip` LIKE CONCAT(?, '%')" : "`ip` = ?";
+	$userlist = $psdb->query(
+		"SELECT `username`, `userid`, `banstate` FROM `ntbb_users` WHERE " . $where,
+		[$ip]
+	);
```

Comment:
```suggestion
	if (substr($ip, -1) === "*") { // ip range
		$ip = substr($ip, 0, strlen($ip) - 1) . '%';
		$userlist = $psdb->query("SELECT `username`, `userid`, `banstate` FROM `ntbb_users` WHERE `ip` LIKE ?", [$ip]);
	} else {
		$userlist = $psdb->query("SELECT `username`, `userid`, `banstate` FROM `ntbb_users` WHERE `ip` = ?", [$ip]);
	}
```

Reply:
In general I prefer for strings passed to `query` to be literals unless _absolutely_ necessary.